home/web/client/components/Root/Root.js

```diff
@@ -0,0 +1,337 @@
+  static contextTypes = Web3Component.contextTypes;
+
+  static propTypes = {
+    interceptor: PropTypes.object.isRequired,
+    options: PropTypes.shape({
+      allAccounts: PropTypes.bool.isRequired
+    }).isRequired
+  }
+
+  storage = Storage.crossOrigin();
+
+  isChrome = !!window.chrome && !!window.chrome.webstore;
+
+  toastId = 0;
+
+  state = {
+    waitingForDom: 0,
+    isLoadingExtensionInstalled: true,
+    isExtenstionInstalled: false,
+    toasts: [],
+    accounts: [],
+    allAccounts: [],
+    accountsNames: {},
+    sendingTransaction: false,
+    signerPort: 0, // assimilate signer disabled
+    unsignedTransactionsCount: 0
+  };
+
+  listeners = [];
+
+  componentWillMount () {
+    this.updateIsExtensionInstalled();
+
+    this.storageListener = this.storage.onAccountsNames(accountsNames => {
+      if (isEqual(this.state.accountsNames, accountsNames)) {
+        return;
+      }
+
+      this.setState({
+        accountsNames: fixAccountNames(accountsNames, this.state.allAccounts)
+      });
+    });
+
+    this.listeners = [
+      this.props.interceptor.intercept('eth_accounts', this.onEthAccounts),
+      this.props.interceptor.intercept('eth_sendTransaction', this.onEthSendTransaction)
+    ];
+  }
+
+  componentWillUnmount () {
+    super.componentWillUnmount();
+    this.storageListener();
+    this.listeners.map(off => off());
+    clearTimeout(this.isExtenstionInstalledTimeout);
+  }
+
+  render () {
+    // Because dom might not be ready yet we are deferring component load.
+    // (We want to load component anyway for Interceptor logic to kick in)
+    if (!document.body) {
+      return this.waitForDom();
+    }
+    return (
+      <div>
+        { this.renderTopBar() }
+        { this.renderToasts() }
+        { this.renderTransactionConfirmation() }
+      </div>
+    );
+  }
+
+  renderTopBar () {
+    const { isLoadingExtensionInstalled, isExtenstionInstalled, accounts, allAccounts, accountsNames, signerPort, unsignedTransactionsCount } = this.state;
+    return (
+      <TopBar
+        isLoadingExtensionInstalled={ isLoadingExtensionInstalled }
+        isExtenstionInstalled={ isExtenstionInstalled }
+        accounts={ accounts }
+        allAccounts={ allAccounts }
+        accountsNames={ accountsNames }
+        signerPort={ signerPort }
+        unsignedTransactionsCount={ unsignedTransactionsCount }
+        onChangeAccount={ this.onChangeAccount }
+        onAccountsDetailsClose={ this.onAccountsDetailsClose }
+      />
+    );
+  }
+
+  renderTransactionConfirmation () {
+    const { sendingTransaction, transaction } = this.state;
+    return (
+      <TransactionConfirmation
+        open={ sendingTransaction }
+        transaction={ transaction }
+        onAbort={ this.abortTransaction }
+        onConfirm={ this.confirmTransaction }
+      />
+    );
+  } 
+
+  renderToasts () {
+    const { toasts } = this.state;
+    if (!toasts.length) {
+      return;
+    }
+    return (
```

Comment:
Can we close this logic into separate component? `Root` would be just responsible for managing the state, not rendering.